### ISSUE TITLE
In freeze, Git tag gets replaced with potentially unusable SHA

### ISSUE THREAD

**Issue by gracinet** [anybox/anybox.recipe.odoo#33]
This is a special case of issue #2 which qualifies as a bug (infrequent)

Actually, even tags used in the buildout to freeze get replaced by their SHA (which at least cannot be garbage collected).

Now, SHAs can be fetched only if they are reachable from the HEAD of remote repo (typically master), which means that currently tags on feature branches cannot be used at all in the freeze/extract part of the process. This is not a projection, it just happened on a real project at Anybox.

**Comment by foutoucour** [comment]
We are facing the same issues at Savoir-faire Linux.
We had to make a wrapper to run a bit of code before to run the freeze command.

The bit of code looks like:
```
git clean -xfd
git reset --hard origin/8.0
```

Hope it helps

**Comment by gracinet** [comment]
@foutoucour thanks for feedback, but I think it's unrelated.

Indeed the commands you're showing actually circumvent the refusal to freeze if local modifications of any kind are found, which is a feature deliberately introduced from the ground up: it protects the user against releasing something that may behave differently than the local copy (a local modification actually leading to a significative behaviour difference is uncommon, but each occurrence has a high cost in terms of debugging time).
So in short, I don't recommend what you're doing, unless one is sure in the given context that any local modifications are superficial. And even in that case, I'd try to circumvent with .gitgnore first.

In the past, it has happened that local modifications could have been introduced by the recipe itself (setup.py) in which case of course nothing becomes releasable, in particular not on bots, and that's been treated as a bug in itself.

Regards,

**Comment by gracinet** [comment]
@sandreanybox would you please be so kind to try the current a.r.openerp-1.9 branch, namely the [8ae5bb6] I just pushed. They should work better for you.
cc @jssuzanne, since we talked about it earlier
